fulcio.proto

```diff
@@ -0,0 +1,120 @@
+option java_package = "dev.sigstore.fulcio.v1";
+option java_multiple_files = true;
+option java_outer_classname = "FulcioProto";
+
+service CA {
+    /**
+     * Returns information about the current state of the transparency log
+     */
+    rpc CreateSigningCertificate(CreateSigningCertificateRequest) returns (SigningCertificate){
+        option (google.api.http) = {
+          post: "/api/v2/signingCert"
+          body: "*"
+        };
+    }
+    /**
+     * Returns the bundle of certificates that can be used to validate the signed certificate timestamp
+     */
+    rpc GetTrustBundle(google.protobuf.Empty) returns (TrustBundle){
+        option (google.api.http) = {
+          get: "/api/v2/trustBundle"
+        };
+    }
+}
+
+message CreateSigningCertificateRequest {
+    /*
+     * Identity information about who possesses the private / public key pair presented
+     */
+    Credentials credentials    = 1 [(google.api.field_behavior) = REQUIRED];
+    /*
+     * The public key to be stored in the requested certificate
+     */
+    PublicKey public_key       = 2 [(google.api.field_behavior) = REQUIRED];
+    /*
+     * Proof that the client possesses the private key; must be verifiable by provided public key
+     */
+    bytes proof_of_possession  = 3 [(google.api.field_behavior) = REQUIRED];
+}
+
+message Credentials {
+    oneof credentials {
+        /*
+        * The OIDC token that identifies the caller
+        */
+        string oidc_identity_token = 1;
+    }
+}
+
+message PublicKey {
+    /*
+     * The cryptographic algorithm to use with the key material
+     */
+    PublicKeyAlgorithm algorithm = 1;
+    /*
+     * PEM encoded public key
+     */
+    string content               = 2;
+}
+
+// (-- api-linter: core::0142::time-field-type=disabled
+//     aip.dev/not-precedent: SCT is defined in RFC6962 and we keep the name consistent for easier understanding. --)
+message SigningCertificate {
+    /*
+     * The signed certificate timestamp is a promise for including the certificate in
+     * a certificate transparency log. It can be "stapled" to verify the inclusion of
+     * a certificate in the log in an offline fashion.
+     */
+    bytes signed_certificate_timestamp = 1;
+    /*
+     * The full certificate chain, serialized with the leaf certificate first, followed
+     * by all intermediate certificates (if present), finishing with the root certificate.
+     *
+     * All values are PEM-encoded certificates.
+     */
+    repeated string certificates = 2;
+}
+
+message TrustBundle {
+    /*
+     * The PEM-encoded root certificate chain for this Fulcio instance
+     */
+    repeated CertificateChain chains = 1;
+}
+
+message CertificateChain {
+    /*
+     * The PEM-encoded root certificate chain for this Fulcio instance
+     */
+    repeated string certificates = 1;
+}
+
+enum PublicKeyAlgorithm {
+    PUBLIC_KEY_ALGORITHM_UNSPECIFIED = 0;
```

Comment:
Question: Do we actually support anything outside ECDSA key-pairs right now? I've seen places around sigstore project where its sort of hard coded to ECDSA and wasn't sure if there is actually flexibility there

Reply:
Right right, is there even a need for this ever? What is the API even supposed to do if there is a mismatch? The alg type is always available in the public key anyways right?